### PR TITLE
Implemented Kafka repository

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- Introduced new request types `BlockGroup` and `TraceGroup`.
- Implemented data transformation before publishing Kafka messages.
- Implemented Kafka repository to implement `HandleChainEvent` interface method.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 1]
```diff
@@ -86,45 +86,54 @@ func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig)
...
-		setter       ComponentSetter
+		setter       []ComponentSetter
 		err          error
 	)
 	switch cfg.Mode {
 	case ModeKAS:
-		newRepo, err := kas.NewRepository(cfg.KasConfig)
+		repo, checkpointDB, setter, err = getKasComponents(cfg.KasConfig)
 		if err != nil {
-			logger.Error("Failed to create new Repository", "err", err, "user", cfg.KasConfig.DBUser, "host", cfg.KasConfig.DBHost, "port", cfg.KasConfig.DBPort, "name", cfg.KasConfig.DBName, "cacheUrl", cfg.KasConfig.CacheInvalidationURL, "x-chain-id", cfg.KasConfig.XChainId)
```
Comment (winnie-byun) [inline datasync/chaindatafetcher/chaindata_fetcher.go:96]: Why did you erased the log?

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 2]
```diff
@@ -86,45 +86,54 @@ func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig)
 	var (
 		repo         Repository
 		checkpointDB CheckpointDB
-		setter       ComponentSetter
+		setter       []ComponentSetter
```
[the remark this answers is not in the document]
Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/chaindata_fetcher.go]: Thanks! 95cf9c89f8324594e19e72d9151e6bdf1b6bdb6c

File: datasync/chaindatafetcher/chaindata_fetcher.go (hunk 1 above)
Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/chaindata_fetcher.go:96]: This is a duplicate log. In the `NewRepository` methods for both KAS and Kafka, there is an error log when the creation is failed.

**Review by yoomee1313 — APPROVED** [review]

**Review by KimKyungup — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 3]
```diff
@@ -86,45 +86,54 @@ func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig)
...
-		checkpoint:    checkpoint,
 		checkpointMap: make(map[int64]struct{}),
 		repo:          repo,
 		checkpointDB:  checkpointDB,
-		setter:        setter,
+		setters:       setters,
 	}, nil
 }
 
+func getKasComponents(cfg *kas.KASConfig) (Repository, CheckpointDB, []ComponentSetter, error) {
```
Comment (ehnuje) [inline datasync/chaindatafetcher/chaindata_fetcher.go:120]: Returning 4 variables looks verbose, please consider refactoring them later.

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 4]
```diff
@@ -287,22 +303,40 @@ func (f *ChainDataFetcher) Components() []interface{} {
 	return nil
 }
 
+func (f *ChainDataFetcher) setAPIs(apis []rpc.API) {
```
Comment (ehnuje) [inline datasync/chaindatafetcher/chaindata_fetcher.go:306]: This only sets `cn.PrivateDebugAPI`, which does not match its name.

Comment (ehnuje) [reply datasync/chaindatafetcher/chaindata_fetcher.go:306]: Is this an interface method?

**Review by ehnuje — APPROVED** [review]
LGTM except for minor comments.

File: datasync/chaindatafetcher/chaindata_fetcher.go (hunk 3 above)
Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/chaindata_fetcher.go:120]: Aright. I will make another PR for this.

File: datasync/chaindatafetcher/chaindata_fetcher.go (hunk 4 above)
Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/chaindata_fetcher.go:306]: Nope, I will make another PR for this as well with the above comment.

**Pull request merged**